From 2e97e467d7ab13a83573e9b48424442f23edc22b Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 02 Jul 2023 19:24:20 +0300
Subject: [PATCH] [SW-150748] make: do not use lto plugin

after upgrading to Synopsys Arc toolchain 2023.03, ar program error
is observed when attempt to use lto plugin. In addition, when
building without lto plugin no change in binary is observed.
Hence, for now, the lto plugin is not used.

Change-Id: I8b439b8ac0293b4f6f01b86feb16d5d5f51240af
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/Makefile b/Makefile
index dfc97bc..1a9e1f4 100644
--- a/Makefile
+++ b/Makefile
@@ -4,11 +4,12 @@
 SPECS_EMB_DEV := $(shell echo $(HL_DEV) | cut -d'_' -f1)
 HL_DEV_SUBTYPE := $(shell echo $(HL_DEV) | cut -d'_' -s -f2,3)
 HL_DEV_ARCH := arch/$(shell echo $(HL_DEV) | cut -d'_' -f1)/$(HL_DEV_SUBTYPE)
-ifneq ($(CROSS_COMPILE),)
-	export LTO_PLUGIN := --plugin $(shell find $(abspath ../../) -name liblto_plugin.so)
-	RANLIB += $(LTO_PLUGIN)
-	export ARFLAGS += $(LTO_PLUGIN)
-endif
+
+# removed liblto_plugin.so plugin from flags.
+# after upgrading to Synopsys Arc toolchain 2023.03, ar error is observed when attempt to use
+# lto plugin. In addition, when building without lto plugin no change in binary is observed.
+# Hence, for now, the lto plugin is not used.
+
 export CC AR RANLIB
 OBJCOPY = $(CROSS_COMPILE)objcopy
 
